spec/criteria_spec.cr

```diff
@@ -44,6 +44,24 @@ describe LuckyRecord::Criteria do
       age.lte("30").to_sql.should eq ["SELECT #{QueryMe::COLUMNS} FROM users WHERE age <= $1", "30"]
     end
   end
+
+  describe "not" do
+    describe "with a value" do
+      it "negates to not equal" do
+        age.not("30").to_sql.should eq ["SELECT #{QueryMe::COLUMNS} FROM users WHERE age != $1", "30"]
+      end
+    end
+
+    describe "without a value" do
+      it "negates the following criteria" do
+        age.not.ilike("3").to_sql.should eq ["SELECT #{QueryMe::COLUMNS} FROM users WHERE age NOT ILIKE $1", "3"]
+      end
+    end
+
+    it "it resets after having negated once" do
```

Comment:
This is a really nice spec. I think it will be great to have in order to be sure we don't have any regressions in the future 👍